adagrams/game.py

```diff
@@ -1,11 +1,119 @@
+"""Adagrams: It's really scrabble"""
+
+import random
+
+LETTER_POOL = {
+    'A': 9,
+    'B': 2,
+    'C': 2,
+    'D': 4,
+    'E': 12,
+    'F': 2,
+    'G': 3,
+    'H': 2,
+    'I': 9,
+    'J': 1,
+    'K': 1,
+    'L': 4,
+    'M': 2,
+    'N': 6,
+    'O': 8,
+    'P': 2,
+    'Q': 1,
+    'R': 6,
+    'S': 4,
+    'T': 6,
+    'U': 4,
+    'V': 2,
+    'W': 2,
+    'X': 1,
+    'Y': 2,
+    'Z': 1
+}
+
+letter_value = {
+    1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+    2: ['D', 'G'],
+    3: ['B', 'C', 'M', 'P'],
+    4: ['F', 'H', 'V', 'W', 'Y'],
+    5: ['K'],
+    8: ['J', 'X'],
+    10: ['Q', 'Z']
+}
+
+
 def draw_letters():
-    pass
+    """Output a list of ten strings, where each string is a letter of 
+    len(1) as defined in LETTER_POOL."""
+
+    letter_pool_copy = LETTER_POOL.copy()
+    letters = []
+
+    while len(letters) < 10:
+        letter_key = random.choice(list(letter_pool_copy.keys()))
+        if letter_pool_copy[letter_key] != 0:
+            letter_pool_copy[letter_key] -= 1
+            letters.append(letter_key)
+
+    return letters
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    """Each letter in the word 1) is not case-sensitive and 2) is not 
+    utilized more than the available frequency outlined in LETTER_POOL."""
+
+    letter_bank_copy = letter_bank.copy()
+    letter_bank_case = [element.upper() for element in letter_bank_copy]
```

Comment:
Consider combining these into one! Line 65 is not making a true copy of `letter_bank`. `copy` makes a "shallow" copy that still points to the original reference IDs of the items _inside_ the list. Let's refactor line 66 a bit:

```python
    letter_bank_copy = [element.upper() for element in letter_bank.deepcopy()]
```
This would now make a "deep" copy of each value in the list, and then convert each value to its uppercase in memory,.